common/remote/dafsstream.cpp

```diff
@@ -0,0 +1,1329 @@
+
+        fieldFilters.push_back(textFilter);
+        variableContentDirty = true;
+    }
+    virtual void clearFieldFilters() override
+    {
+        fieldFilters.clear();
+        variableContentDirty = true;
+    }
+    virtual void setOutputRecordFormat(const char *eclRecDef) override
+    {
+        MultiErrorReceiver errs;
+        Owned<IHqlExpression> record = parseQuery(eclRecDef, &errs);
+        if (errs.errCount())
+        {
+            StringBuffer errText;
+            IError *first = errs.firstError();
+            first->toString(errText);
+            throwDsFsClientExceptionV(DaFsClient_ECLParseError, "Failed to parse output ecl definition '%s': %s @ %u:%u", eclRecDef, errText.str(), first->getColumn(), first->getLine());
+        }
+        if (!record)
+            throwDsFsClientExceptionV(DaFsClient_ECLParseError, "Failed to parse output ecl definition '%s'", eclRecDef);
+
+        MemoryBuffer projectedTypeInfo;
+        exportBinaryType(projectedTypeInfo, record, dft_index == file->queryType());
+        outMeta.setown(createTypeInfoOutputMetaData(projectedTypeInfo, false)); // JCSMORE if file->queryIsGrouped(), should this be? and if so, ctor outMeta needs reconstructing
+        StringBuffer typeInfoStr;
+        JBASE64_Encode(projectedTypeInfo.toByteArray(), projectedTypeInfo.length(), typeInfoStr, false);
+        requestNode->setProp("outputBin", typeInfoStr.str());
+    }
+    virtual void addVirtualFieldMapping(const char *fieldName, const char *fieldValue) override
+    {
+        variableContentDirty = true;
+        virtualFields[fieldName] = fieldValue;
+    }
+};
+
+
+class CDFUPartWriterBase : public CDaFileSrvClientBase, implements IDFUFilePartWriter
+{
+    typedef CDaFileSrvClientBase PARENT;
+
+    MemoryBuffer replyMb;
+    unsigned startPos = 0;
+    Owned<ICompressor> compressor;
+
+protected:
+    bool firstSend = true;
+    const unsigned sendThresholdBytes = 0x100000; // 1MB
+
+    void prepNext()
+    {
+        sendMb.clear();
+        // prepare for next continuation
+        initSendBuffer(sendMb);
+        Owned<IPropertyTree> tree = createPTree();
+        tree->setPropInt("handle", handle);
+        addRequest(tree);
+        startPos = sendMb.length();
+        size32_t rowDataSz = 0;
+        sendMb.append(rowDataSz); // place holder
+        if (compressor)
+        {
+            void *rowData = sendMb.reserveTruncate(sendThresholdBytes);
+            compressor->open(rowData, sendThresholdBytes);
+        }
+    }
+    unsigned send(MemoryBuffer &replyMb)
+    {
+        size32_t len;
+        if (compressor)
+        {
+            compressor->close();
+            len = compressor->buflen();
+            sendMb.setLength(startPos + sizeof(len)+len);
+        }
+        else
+            len = sendMb.length()-startPos;
+        sendMb.writeEndianDirect(startPos, sizeof(len), &len);
+        return PARENT::send(replyMb);
+    }
+    void _sendWriteFirst()
```

Comment:
let's rename these 2 methods ( _sendWriteFirst and/or sendWriteFirst) so their purpose is clear from the name...

Reply:
done